providers/base/bin/prime_offload_tester.py

```diff
@@ -0,0 +1,334 @@
+                    self.logger.info("  Offload process:[{}]".format(cmd))
+                    self.logger.info("  Card ID:[{}]".format(card_id))
+                    self.logger.info("  Device Name:[{}]".format(card_name))
+                    self.check_result = PrimeOffloaderError.NO_ERROR
+                    return
+                self.check_result = PrimeOffloaderError.OFFLOAD_FAIL
+            except (OSError, TypeError):
+                # Missing file or permissions?
+                self.logger.info("Couldn't open file for"
+                                 "reading clients of dri device")
+                self.check_result = PrimeOffloaderError.OFFLOAD_FAIL
+                return
+        self.logger.info("Checking fail:")
+        self.logger.info("  Couldn't find process [{}]".format(cmd) +
+                         " running after check {} times".format(index))
+        self.check_result = PrimeOffloaderError.OFFLOAD_FAIL
+
+    def check_nv_offload_env(self):
+        """
+        prime offload of nvidia driver is limited.
+        Only on-demand mode is supported.
+        """
+        # nvidia-smi ship with NVIDIA GPU display drivers on Linux
+        # https://developer.nvidia.com/nvidia-system-management-interface
+        # check prime-select to make sure system with nv driver.
+        # If there is no nv driver, prime offload is fine for other drivers.
+        ps = subprocess.run("whereis prime-select | cut -d ':' -f 2",
+                            shell=True,
+                            stdout=subprocess.PIPE,
+                            universal_newlines=True)
+        if 'prime-select' not in ps.stdout:
+            return PrimeOffloaderError.NO_ERROR
+
+        # prime offload could running on on-demand mode only
+        mode = subprocess.run("prime-select query",
+                              shell=True,
+                              stdout=subprocess.PIPE,
+                              universal_newlines=True)
+        if "on-demand" not in mode.stdout:
+            self.logger.info("System isn't on-demand mode")
+            return PrimeOffloaderError.NOT_SUPPORT_NV_PRIME
+
+        # prime offload couldn't running on nvlink active or inactive
+        # Therefore, only return empty string is supported environment.
+        nvlink = subprocess.run("nvidia-smi nvlink -s",
+                                shell=True,
+                                stdout=subprocess.PIPE,
+                                universal_newlines=True)
+        if len(nvlink.stdout) != 0:
+            if re.search('error', nvlink.stdout, re.IGNORECASE):
+                self.logger.info("nvidia driver error")
+                return PrimeOffloaderError.NV_DRIVER_ERROR
+            self.logger.info("NVLINK detected")
+            return PrimeOffloaderError.NOT_SUPPORT_NV_PRIME
+
+        return PrimeOffloaderError.NO_ERROR
+
+    def run_offload_cmd(self, cmd, pci_name, driver, timeout):
```

Comment:
if the callers now about supplying timeouts, they as well may provide the right cmd (with timeout).